src/main.cpp

```diff
@@ -3069,9 +3119,18 @@ bool CheckBlock(const CBlock& block, CValidationState& state,
             return state.DoS(100, error("CheckBlock(): more than one coinbase"),
                              REJECT_INVALID, "bad-cb-multiple");
 
+    // FIXME: Need test for CheckBlock 
+    // Get branch id based on height of block which should advance tip
+    uint32_t consensusBranchId = 0;
+    int futureBlockHeight = chainActive.Height() + 1;
```

Comment:
This will not be correct for upgrades after Overwinter - it will result in Overwinter blocks being checked against the chain tip's consensus rules. We should either pass the block's block height into `CheckBlock()`, or (more likely) pass `consensusBranchId` in.